src/main/java/org/fcrepo/importexport/common/BagConfig.java

```diff
@@ -138,12 +123,34 @@ private void validateAPTrust(final Map<String, String> apTrustInfo) {
     }
 
     /**
-     * Returns an immutable map of custom tags.
+     * Returns all the tag files from the config
+     *
+     * @return set of tag filenames
+     */
+    public Set<String> getTagFiles() {
+        return map.keySet();
+    }
+
+    /**
+     * Check if a tag file is listed in bag config
+     *
+     * @param tagFile the tag filename
+     * @return true if it is list, false if not
+     */
+    public boolean hasTagFile(final String tagFile) {
+        return map.containsKey(tagFile);
+    }
+
+    /**
+     * Returns an immutable map of custom tags for a tag file
+     *
+     * @param tagFile name of the tag file to get fields for
      * @return a map of filenames to key-value property maps
      */
-    public Map<String, Map<String, String>> getCustomTags() {
-        final Map<String, Map<String, String>> clone = new HashMap(map);
-        clone.remove(BAG_INFO_KEY);
-        return Collections.unmodifiableMap(clone);
+    public Map<String, String> getFieldsForTagFile(final String tagFile) {
+        if (map.get(tagFile) != null) {
```

Comment:
Wouldn't `map.get(tagFile)` return `null` if we just skipped the check?  Is there something else going on here?